x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -98,77 +141,169 @@ export const signalRulesAlertType = ({
-          interval,
-          enabled,
-          pageSize: searchAfterSize,
-          tags,
+          size: searchAfterSize,
+          searchAfterSortId: undefined,
         });
 
-        if (bulkIndexResult) {
+        try {
+          logger.debug(
+            `Starting signal rule name: "${name}", id: "${alertId}", rule_id: "${ruleId}"`
+          );
           logger.debug(
-            `Finished signal rule name: "${name}", id: "${alertId}", rule_id: "${ruleId}"`
+            `[+] Initial search call of signal rule name: "${name}", id: "${alertId}", rule_id: "${ruleId}"`
           );
-        } else {
+          const noReIndexResult = await services.callCluster('search', noReIndex);
+          if (noReIndexResult.hits.total.value !== 0) {
+            logger.info(
+              `Found ${
+                noReIndexResult.hits.total.value
+              } signals from the indexes of "[${inputIndex.join(
+                ', '
+              )}]" using signal rule name: "${name}", id: "${alertId}", rule_id: "${ruleId}", pushing signals to index "${outputIndex}"`
+            );
+          }
+
+          const bulkIndexResult = await searchAfterAndBulkCreate({
+            someResult: noReIndexResult,
+            ruleParams: params,
+            services,
+            logger,
+            id: alertId,
+            signalsIndex: outputIndex,
+            filter: esFilter,
+            name,
+            createdBy,
+            updatedBy,
+            interval,
+            enabled,
+            pageSize: searchAfterSize,
+            tags,
+          });
+
+          if (bulkIndexResult) {
+            logger.debug(
+              `Finished signal rule name: "${name}", id: "${alertId}", rule_id: "${ruleId}"`
+            );
+            const sDate = new Date().toISOString();
+            currentStatusSavedObject.attributes.status = 'succeeded';
+            currentStatusSavedObject.attributes.statusDate = sDate;
+            currentStatusSavedObject.attributes.lastSuccessAt = sDate;
+            currentStatusSavedObject.attributes.lastSuccessMessage = 'succeeded';
+            await services.savedObjectsClient.update(
+              ruleStatusSavedObjectType,
+              currentStatusSavedObject.id,
+              {
+                ...currentStatusSavedObject.attributes,
+              }
+            );
+          } else {
+            logger.error(
+              `Error processing signal rule name: "${name}", id: "${alertId}", rule_id: "${ruleId}"`
+            );
+            const sDate = new Date().toISOString();
+            currentStatusSavedObject.attributes.status = 'failed';
+            currentStatusSavedObject.attributes.statusDate = sDate;
+            currentStatusSavedObject.attributes.lastFailureAt = sDate;
+            currentStatusSavedObject.attributes.lastFailureMessage = `Bulk Indexing signals failed. Check logs for further details \nRule name: "${name}"\nid: "${alertId}"\nrule_id: "${ruleId}"\n`;
```

Comment:
For recording failure messages I would not put any formatting in, we can iterate on more of this later you don't have to change it here. If anything this might become really quickly more structured so that we can do error messages front end with i18n, vs just displaying what the server says. Or at least part of it translatable.

For now, we will just roll forward with what we have.

Reply:
I agree. I would like to return an object with more details from the bulk create method. This would provide helpful messaging in this case.